### PR TITLE
Madison - enforce game in progress for MilkTheCowsJob and InstructorReportJob

### LINKED ISSUE #34: Add logic to the backends of various jobs that operate on all commons (MilkTheCows, UpdateCowHealth, InstructorReports) where it iterates over the commons.  Skip over commons where gameInProgress() returns false.  You can find some code for this in a PR that was not merged from S23.

### LINKED ISSUE #35: If instructor requests a job be run on a specific commons, it should run even if the game is not in progress.
Extension of priority issue 4 future issues

### PR DESCRIPTION
**Pull request by madisonlong1**
## Overview
This pull request
- completes issue 35, " instructor requests a job be run on a specific commons, it should run even if the game is not in progress."
- works on issue 34, Add logic to the backends of various jobs that operate on all commons (MilkTheCows, UpdateCowHealth, InstructorReports) where it iterates over the commons. Skip over commons where gameInProgress() returns false
NOTE: this PR will complete the implementation of adding gameInProgress() logic to MilkTheCowsJob.java and InstructorReportsJob.java, but NOT UpdateCowHealthJob.java. Professor Conrad stated this element could be left for W24

Deployed on Dokku:
https://happycows-madisonlong1.dokku-03.cs.ucsb.edu/

## Screenshots (Optional)
For a game that has not started yet
image

Now, instructorReports and MilkTheCows will not run if the game is not currently in progress, not as per the issue description, this was a **backend** change and is **not** currently visable from the frontend
image

## Validation (Optional)

1. Download this branch.
2. Run the project.
3. Go to this page.
4. Test this function.

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #34
Closes #35

### CODE REVIEW

File: frontend/src/main/components/Jobs/UpdateCowHealthForm.js
Comment (wyatthspivak) [inline frontend/src/main/components/Jobs/UpdateCowHealthForm.js:1]: Possibly delete unnecessary newlines

File: src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java
Comment (wyatthspivak) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java:1]: Possibly delete unnecessary newlines

File: src/test/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobTests.java
Comment (wyatthspivak) [inline src/test/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobTests.java:1]: Possibly delete unnecessary newlines

File: src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJob.java
Comment (wyatthspivak) [inline src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJob.java:1]: Delete unnecessary comment

**Review by wyatthspivak — COMMENTED** [review]
LGTM except for minor changes I listed

File: src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java
Comment (madisonlong1) [reply src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java:1]: done ;)

File: src/test/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobTests.java
Comment (madisonlong1) [reply src/test/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJobTests.java:1]: deleted new lines

File: frontend/src/main/components/Jobs/UpdateCowHealthForm.js
Comment (madisonlong1) [reply frontend/src/main/components/Jobs/UpdateCowHealthForm.js:1]: Removed newlines in UCHF.js :)

**Review by wyatthspivak — DISMISSED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJob.java [hunk 1]
```diff
@@ -29,24 +28,18 @@ public class UpdateCowHealthJob implements JobContextConsumer {
...
-
         Iterable<Commons> allCommons = commonsRepository.findAll();
         Iterable<CommonsPlus> allCommonsPlus = commonsPlusBuilderService.convertToCommonsPlus(allCommons);
 
         for (CommonsPlus commonsPlus : allCommonsPlus) {
-
-
             Commons commons = commonsPlus.getCommons();
-            
+            //ctx.log(String.format("commoms.startDate=%s commons.lastDay=%s", commons.getStartingDate().toString(),commons.getLastDay().toString()));
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJob.java]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: src/test/java/edu/ucsb/cs156/happiercows/jobs/InstructorReportJobTests.java [hunk 2]
```diff
@@ -60,4 +62,71 @@ void test_log_output() throws Exception {
...
+        when(commonsRepository.findAll()).thenReturn(Arrays.asList(commons));      
+        when(reportService.createReport(17L)).thenReturn(report);
+
+        // Act
+        InstructorReportJob instructorReportJob = new InstructorReportJob(reportService, commonsRepository);
+        instructorReportJob.accept(ctx);
+
+        // Assert
+
+        // verify(commonsRepository).findAll();
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/jobs/InstructorReportJobTests.java]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: src/test/java/edu/ucsb/cs156/happiercows/jobs/InstructorReportJobTests.java [hunk 3]
```diff
@@ -60,4 +62,71 @@ void test_log_output() throws Exception {
...
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/jobs/InstructorReportJobTests.java]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by pconrad — DISMISSED** [review]
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Comment by madisonlong1** [comment]
> https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

Comments removed

**Review by kaitntom — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**